src/app/Heading.tsx

```diff
@@ -3,16 +3,31 @@ import React from "react";
 import Steps from "./Steps";
 import Box, { BoxProps } from "@mui/material/Box";
 import Panel from "../components/Panel";
-import { useSelector } from "./AccessibleForm";
+import { useSelector, useDispatch } from "./StoreProvider";
 
 const Heading: React.FC<BoxProps> = (props) => {
   const activeStep = useSelector((state) => state.step);
+  const dispatch = useDispatch();
   return (
     <Box display="flex" justifyContent="space-between" width="100%" {...props}>
       <div css={{ width: "10%" }}>
         <img src="/logo.svg" alt="logo of FormA11y" />
       </div>
-      <Steps width="80%" activeStep={activeStep} />
+      <Steps
+        width="80%"
+        onStepChange={(step) => {
+          dispatch({
+            type: "SET_STEP",
+            payload: step,
+          });
+          // We also need to set the tool to the default tool for the new step
```

Comment:
Instead of dispatching two actions, could we make changing the tool part of the logic in the `SET_STEP` reducer?

Reply:
Great suggestion! Fixed it :)